src/server/transaction.cc

```diff
@@ -894,6 +886,24 @@ void Transaction::Conclude() {
   Execute(std::move(cb), true);
 }
 
+void Transaction::EnableShards(std::optional<ShardId> sid) {
```

Comment:
Maybe split this function into:

`EnableShard(ShardId)`
and 
`EnableAllShards()`

That way it's not confusing when you call 

`EnableShard(0)` 

And it's clear when you want all of them:

`EnableAllShards()`